### PR TITLE
fix: do not report on-function-explicit-return-type error when using immer

### PR DESCRIPTION
**Pull request by jsaguet**
The selector was not correct and reported an error when using immer produce function.

### CODE REVIEW

File: tests/rules/on-function-explicit-return-type.test.ts [hunk 1]
```diff
@@ -24,6 +24,12 @@ ruleTester().run(ruleName, rule, {
       on(increment, incrementFunc),
       on(increment, s => incrementFunc(s)),
       on(increment, (s): State => incrementFunc(s)),
+      on(
```
Comment (timdeschryver) [inline tests/rules/on-function-explicit-return-type.test.ts:27]: Can this be a separate test please?

Comment (jsaguet) [reply tests/rules/on-function-explicit-return-type.test.ts:31]: done

**Comment by jsaguet** [comment]
Which behavior do you expect with ngrx-etc? Error or pass?
Currently, if I'm not mistaken, it should pass since you're either using `mutableOn` or `createMutableReducer` and the rule looks for both `createReducer`AND `on`

**Comment by timdeschryver** [comment]
It should pass.
🤦‍♂️ you're right, which is great as we don't have to do anything for it 🎉

For more context: It's in the TSLint repo where some rules are looking for `on` and `mutableOn`...
Thanks! LGTM

**Pull request merged**